### PR TITLE
Http2 noalloc header compare 5454 v2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5454

Describe changes:
- http2: remove `to_vec` in comparisons (no need to alloc)

@jasonish does clippy catch this ?

Replaces #7666 with version compatible with rust MSRV 1.48 and adds a commit for CI to test MSRV

### CODE REVIEW

File: rust/src/http2/decompression.rs [hunk 1]
```diff
@@ -168,13 +168,13 @@ impl HTTP2DecoderHalf {
     pub fn http2_encoding_fromvec(&mut self, input: &Vec<u8>) {
```
Comment (jasonish) [inline rust/src/http2/decompression.rs:168]: Clippy will usually complain about `&Vec` in signatures, especially if a `&[u8]` could be used instead.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 8437

File: rust/src/http2/decompression.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/http2/decompression.rs:168]: I meant : does clippy complain about running `to_vec` to do a comparison ?

Comment (jasonish) [reply rust/src/http2/decompression.rs:168]: > I meant : does clippy complain about running `to_vec` to do a comparison ?

Not that I can tell.

Comment (catenacyber) [reply rust/src/http2/decompression.rs:168]: Should it not ?
Should we ask someone ?

Comment (jasonish) [reply rust/src/http2/decompression.rs:168]: I'm not seeing a lint that would be called out for this, or one of our supressiosn is also suppressing this.  I suppose we could hunt that down and contribute a lint?  But I think looking at the function signature first, and not using `&Vec` would have resulted in a better choice for the comparison to begin with.

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7685

File: rust/src/http2/decompression.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/http2/decompression.rs:168]: Here is a minimal reproducer

```
fn main() {
    let a = vec![0x33, 0x34, 0x35];
// bad
    if a == "345".as_bytes().to_vec() {
        println!("Hello, world!");
    }
// good
    if &a == "345".as_bytes() {
        println!("Hello, world!");
    }
}
```

**Pull request closed without merging**